src/lazy/expanded/e_expression.rs

```diff
@@ -56,35 +44,40 @@ impl<'top, D: Decoder> ArgGroup<'top, D> {
     pub fn raw_arg_group(&self) -> <D::EExp<'top> as RawEExpression<'top, D>>::ArgGroup {
         self.raw_arg_group
     }
-    pub fn invoked_macro(&self) -> MacroRef<'top> {
-        self.invoked_macro
-    }
-    pub fn expressions(&self) -> ArgGroupIterator<'top, D> {
-        ArgGroupIterator::new(self.context, self.raw_arg_group())
+    pub fn expressions(&self) -> EExpArgGroupIterator<'top, D> {
+        EExpArgGroupIterator::new(self.context, self.raw_arg_group())
     }
-    pub fn expand(&self, environment: Environment<'top, D>) -> IonResult<MacroExpansion<'top, D>> {
+    pub fn expand(&self) -> IonResult<MacroExpansion<'top, D>> {
         let context = self.context();
-        let arguments = MacroExprArgsIterator::from_arg_group(self.expressions());
-        let expansion_kind = MacroExpansionKind::Values(ValuesExpansion::new(arguments));
-        Ok(MacroExpansion::new(context, environment, expansion_kind))
+        let arguments = MacroExprArgsIterator::from_eexp_arg_group(self.expressions());
+        let expansion_kind = MacroExpansionKind::ExprGroup(ExprGroupExpansion::new(arguments));
+        Ok(MacroExpansion::new(
+            context,
+            Environment::empty(),
+            expansion_kind,
+        ))
     }
 }
 
-impl<'top, D: Decoder> HasRange for ArgGroup<'top, D> {
+impl<'top, D: Decoder> HasRange for EExpArgGroup<'top, D> {
     fn range(&self) -> Range<usize> {
         self.raw_arg_group.range()
     }
 }
 
-impl<'top, D: Decoder> HasSpan<'top> for ArgGroup<'top, D> {
+impl<'top, D: Decoder> HasSpan<'top> for EExpArgGroup<'top, D> {
     fn span(&self) -> Span<'top> {
         self.raw_arg_group.span()
     }
 }
 
-impl<'top, D: Decoder> Debug for ArgGroup<'top, D> {
+impl<'top, D: Decoder> Debug for EExpArgGroup<'top, D> {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(f, "ArgGroup {:?}", self.raw_arg_group)
+        write!(f, "(: {:?}", self.raw_arg_group)?;
+        for expr in self.expressions() {
+            write!(f, " {:?}", expr)?;
+        }
+        write!(f, ")")
```

Comment:
🗺️ The derived `Debug` impl gets very noisy if there are several layers of nesting. I rewrote some of these to make the output more compact and human friendly.